### DISCUSSION TITLE
Future of sysfs functions

### DISCUSSION THREAD

**Discussion by mstroh76** [WiringPi/WiringPi#202] [General]
The deprecated sysfs ABI https://www.kernel.org/doc/Documentation/gpio/sysfs.txt is still used from WiringPi.

Raspberry Pi OS is using Kernel 6.1 and is still supporting sysfs on all models

Ubuntu 24.04 will use 6.8, could be a problem (can't check until released)

**Comment by uugear** [comment]
One hour ago I just did "sudo apt update" and "sudo apt upgrade" in Raspberry Pi OS, and it upgrades the kernel to 6.6.20.
As far as I can see, the "gpio" utility 's "wfi" functionality doesn't work now.
```
pi@raspberrypi:~ $ gpio -g wfi 4 both
gpio: Unable to open GPIO direction interface for pin 4: No such file or directory
wiringPiISR: unable to open /sys/class/gpio/gpio4/value: No such file or directory
```

**Reply by mstroh76** [reply]
😱 ist this true the released kernel 6.6 ?

**Reply by uugear** [reply]
The kernel is really 6.6 now.
```
pi@raspberrypi:~ $ uname -ra
Linux raspberrypi 6.6.20+rpt-rpi-v8 #1 SMP PREEMPT Debian 1:6.6.20-1+rpt1 (2024-03-07) aarch64 GNU/Linux
```
So the sysfs has been removed:
```
pi@raspberrypi:~ $ ls /sys/class/gpio/
export  gpiochip512  gpiochip566  gpiochip568  unexport
```

**Reply by mstroh76** [reply]
😩 no time to rest, 
ISR and 'gpio wfi' will not work until code is rewritten.
Ubuntu already removed sysfs in 23.10 Kernel 6.5

**Reply by mwallner** [reply]
PR #205 has been merged and will soon be available in [WiringPi 3.2](https://github.com/WiringPi/WiringPi/releases/tag/3.2).

**Comment by mwallner** [comment]
sysfs no longer is needed as of [release 3.2](https://github.com/WiringPi/WiringPi/releases/tag/3.2).

**Comment by GerritPolder** [comment]
I just compiled and installed the latest wiringPi version (3.14) and recompiled my own program (433MHz remote control).
The program works fine, but it outputs: 

```
gpio: GPIO Sysfs Interface for Userspace is deprecated (https://www.kernel.org/doc/Documentation/gpio/sysfs.txt).
 Function is now useless and empty.
```

how can I get rid of that message?

**Reply by mstroh76** [reply]
It would be better to start a new Issue for that topic. It's a closed issue! 
sys was removed from kernel.

_gpio export
gpio unexport_
...

This commands are useless and empty, please remove that commands.

**Reply by GerritPolder** [reply]
_gpio export_ wasn’t part of my code, so I was a bit confused at first — but apparently after a reboot, the message disappeared on its own.
